### PR TITLE
Save ports as list rather than string

### PR DESCRIPTION
**Pull request by ramizdundar**
This PR:

- Holds ports as list of integer rather than comma separated string
- Generates client protocol related files
- Holds the list as volatile

Succeeds: https://github.com/hazelcast/hazelcast/pull/23231

Related: https://hazelcast.atlassian.net/wiki/spaces/HZC/pages/4361355275/TPC-Aware+Client+Configuration+TDD#Authentication

### CODE REVIEW

**Comment by pveentjer** [comment]
Looks good to me.

File: hazelcast/src/main/java/com/hazelcast/ClientMain.java [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class ClientMain {
 
     public static void main(String[] args) {
         System.setProperty("hazelcast.tpc.enabled", "true");
-        System.setProperty("hazelcast.tpc.eventloop.count", "" + Runtime.getRuntime().availableProcessors());
+        System.setProperty("hazelcast.tpc.eventloop.count", "" + 10);
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/ClientMain.java:15]: minor: why do we change that?

**Review by mdumandag — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/ClientMain.java (hunk 1 above)
Comment (ramizdundar) [reply hazelcast/src/main/java/com/hazelcast/ClientMain.java]: https://github.com/hazelcast/hazelcast/pull/23253/commits/695de1aef72f18ec77dc2152e1a1f91dae40ad80

Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/ClientMain.java]: To make sure that you run as many TPC threads as there are cores.

**Pull request merged**